### DISCUSSION TITLE
On-demand ISR with hostname-rewrites not revalidating

### DISCUSSION THREAD

**Discussion by ghost** [vercel/community#710] [Help] [answered]
## When trying to revalidate the page on demand by calling revalidate() it always fails

**Simple repository to reproduce the error**
https://github.com/ianmorais25/hostname-rewrites-isr

https://<your-site.com>/api/revalidate?secret=token
-->
Internal Server Error  
Error: Failed to revalidate /: Invalid response 404

**Comment by amyegan** [comment, marked as answer]
Hi @ianmorais25. Revalidate is failing because of a conflict between your middleware setup and the path you're providing to revalidate. If I change [line 5](https://github.com/ianmorais25/hostname-rewrites-isr/blob/c180852efbe13f3e7a407fb2b6991499f0c36d89/pages/api/revalidate.js#L5) to `await res.revalidate("/_sites/subdomain-2");` then it works for me.

If you need more in-depth help with your Next.js setup, [Next.js Discord](https://nextjs.org/blog/discord) or [Next.js GitHub Discussions](https://github.com/vercel/next.js/discussions) will be able to give you more specific  assistance.
